src/App.jsx

```diff
@@ -14,33 +16,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_GOODS_NAME = 'name';
+const SORT_GOODS_LENGTH = 'length';
+const GOODS_REVERSE = 'reverse';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getSortedGoods(goods, sortMethod, query) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortMethod) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortMethod) {
+        case SORT_GOODS_NAME:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        case SORT_GOODS_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (query !== '') {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState('');
```

Comment:
More descriptive name can be used here, also better use boolean value for state here.
Example:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```